quinn-proto/src/connection/assembler.rs

```diff
@@ -59,6 +59,37 @@ impl Assembler {
         Some((n, data))
     }
 
+    // Get the the next ordered chunk
+    pub(crate) fn read_chunk(&mut self) -> Result<Option<Bytes>, IllegalOrderedRead> {
+        if let State::Unordered { .. } = self.state {
+            return Err(IllegalOrderedRead);
+        }
+
+        loop {
+            let chunk = match self.data.peek() {
+                Some(chunk) => chunk,
+                None => return Ok(None),
+            };
+
+            if chunk.offset > self.bytes_read {
+                // Next chunk is after current read index
+                return Ok(None);
```

Comment:
Isn't `None` end of stream? i'm not sure how this function is used, but it might need to return a blocked status instead.

Reply:
This method works as it's siblings: it returns `None` when there is nothing to *assemble*. Blocked status is managed at the *stream* level in `quinn-proto::connection::streams::read_chunk()`.

See the `chunks_dedup()` test at the end of this file for an example of expected behavior.